src/app/integrations/providers/aws-sso.service.ts

```diff
@@ -144,6 +110,12 @@ export class AwsSsoService extends NativeService {
 
     return fromPromise(this.ssooidc.createToken(createTokenRequest).promise()).pipe(
       catchError((err) => {
+        if (err.code === 'AuthorizationPendingException') {
+          // Retry after 5 seconds if token is not yet ready
+          return timer(5000).pipe(
```

Comment:
Is polling the best integration we can do with the external opening window, we are some kind of losing control of what happens in the flow.

Should be a good idea to add a maximum number of retry to manage this integration I think.

Reply:
As I understand it the request times out on AWS side and the error will change.

Reply:
You are right, this use-case should be already managed